ocaml/xapi/xapi_stats.ml

```diff
@@ -35,7 +35,18 @@ let generate_master_stats ~__context =
         ~units:"tasks" ()
     )
   in
-  [session_count_ds; task_count_ds]
+  let total_session_count =
+    Atomic.get Xapi_session.total_sessions |> Int64.of_int
+  in
+  let total_session_count_ds =
+    ( Rrd.Host
+    , Ds.ds_make ~name:"pool_total_session_count"
+        ~description:"Total number of sessions"
+        ~value:(Rrd.VT_Int64 total_session_count) ~ty:Rrd.Gauge ~default:true
```

Comment:
```suggestion
        ~value:(Rrd.VT_Int64 total_session_count) ~ty:Rrd.Derive ~default:true
```

We need the RRds to calculate the rate of change, gauge is time-independent so it doesn't do what we want here. Note that the number of tasks and sessions are the ones that are open at a time instead of the total amount, they are different so a rate cannot be calculated from them.

```
(** DS - a data source
    This defines how we deal with incoming data. Type is one of:
    - Absolute: meaning that the incoming data is an absolute rate
    - Derive:   meaning that the rate must come from the difference between the
                incoming data and the previous value
    - Gauge:    meaning that the value isn't a rate at all (e.g. temperature, load avg)
```
https://github.com/xapi-project/xcp-rrd/blob/master/lib/rrd.ml#L98-107